### ISSUE TITLE
shutdown() does not gracefully drain / clear generic pools

### ISSUE THREAD

**Issue by rossj** [guidesmiths/rascal#61]
Related to #50. I've noticed that after trying to do a graceful `shutdown()`, my process hangs around for about ~30 seconds. Using [wtfnode](https://github.com/myndzi/wtfnode) it seems that there is still a timer from generic-pool that keeps the process running for a while:

```
[WTF Node?] open handles:
- File descriptors: (note: stdio always exists)
  - fd 1 (tty) (stdio)
  - fd 2 (tty) (stdio)
- Timers:
  - (1000 ~ 1000 ms) (anonymous) @ /code/node_modules/generic-pool/lib/generic-pool.js:256
```

I'm guessing this is part of generic-pool's idle handling. `idleTimeoutMillis` defaults to 30s which is about the time I'm seeing before actual process exit.

Perhaps the rascal vhost should drain and clear the pools during shutdown?

From the generic-pool docs:
```javascript
/**
 * Step 3 - Drain pool during shutdown (optional)
 */
// Only call this once in your application -- at the point you want
// to shutdown and stop using this pool.
myPool.drain().then(function() {
  myPool.clear();
});
```

I'm new to rascal but I'd be happy to work on a PR if you'd like this change.

**Comment by cressie176** [comment]
Hi @rossj. Thanks for diagnosing this and the offer of a PR. The channel pooling code is a little ugly, so I'll sort shortly.

**Comment by cressie176** [comment]
Should be fixed in rascal@4.6.0

**Comment by rossj** [comment]
Cool, my process exits immediately now. Thanks!
